test/test.js

```diff
@@ -1433,9 +1433,29 @@ describe('Chat', function() {
 				const message = {
 					id,
 					text: 'yo',
+					user_id: uuidv4(),
 				};
-				await channel.sendMessage(message);
-				await expectHTTPErrorCode(400, channel.sendMessage(message));
+				const client = await getTestClient(true);
+				let serverChannel = client.channel(channel.type, channel.id);
+				await serverChannel.sendMessage(message);
+				await expectHTTPErrorCode(
+					400,
+					serverChannel.sendMessage(message),
+					`StreamChat error code 4: SendMessage failed with error: "a message with ID ${id} already exists"`,
+				);
+			});
+
+			it('Add a chat message with ID client side should fail', async function() {
```

Comment:
@thesyncim we generate id on client side in react and RN, before sending a message. Are you sure it should fail?

https://github.com/GetStream/stream-chat-react/blob/master/src/components/Channel.js#L335